packages/venia-concept/src/ServiceWorker/Utilities/htmlHandler.js

```diff
@@ -1,7 +1,144 @@
+const generateResources = parsedHTML => {
+    const resources = {};
+
+    resources.scripts = parsedHTML
+        .querySelectorAll('script')
+        .map(generateScriptResource)
+        .filter(identity)
+        .sort();
+
+    resources.links = parsedHTML
+        .querySelectorAll('link')
+        .map(generateLinkResource)
+        .filter(identity)
+        .sort();
+
+    resources.styles = parsedHTML
+        .querySelectorAll('style')
+        .map(generateStyleResource)
+        .filter(identity)
+        .sort();
+
+    return resources;
+};
+
+/**
+ * Returns true if resouce url/urls have changed.
+ *
+ * @param {HTMLElement} newDOM
+ * @param {HTMLElement} oldDOM
+ *
+ * @returns Boolean
+ */
+const haveResourcesChanged = (newDOM, oldDOM) => {
+    const newResources = generateResources(newDOM);
+    const oldResources = generateResources(oldDOM);
+    const keysToCompare = Object.keys(newResources);
+
+    return !keysToCompare.reduce((acc, key) => {
+        const hasResourceChanged =
+            newResources[key].toString() === oldResources[key].toString();
+
+        return acc && hasResourceChanged;
+    }, true);
+};
+
+/**
+ * Returns true if title has changed.
+ *
+ * @param {HTMLElement} newDOM
+ * @param {HTMLElement} oldDOM
+ *
+ * @returns Boolean
+ */
+const hasTitleChanged = (newDOM, oldDOM) => {
+    const oldTitle = (oldDOM.querySelector('title') || {}).text;
+    const newTitle = (newDOM.querySelector('title') || {}).text;
+
+    return newTitle !== oldTitle;
+};
+
+/**
+ * Returns true if media backend url has changed.
+ *
+ * @param {HTMLElement} newDOM
+ * @param {HTMLElement} oldDOM
+ *
+ * @returns Boolean
+ */
+const hasBackendUrlChanged = (newDOM, oldDOM) => {
+    const oldUrl = oldDOM.querySelector('html').attributes[
+        'data-media-backend'
+    ];
+    const newUrl = newDOM.querySelector('html').attributes[
+        'data-media-backend'
+    ];
+
+    return newUrl !== oldUrl;
+};
+
+/**
+ * Returns true if image optimizing origin has changed.
+ *
+ * @param {HTMLElement} newDOM
+ * @param {HTMLElement} oldDOM
+ *
+ * @returns Boolean
+ */
+const hasImageOptimizingOriginChanged = (newDOM, oldDOM) => {
+    const oldOrigin = oldDOM.querySelector('html').attributes[
+        'data-image-optimizing-origin'
+    ];
+    const newOrigin = newDOM.querySelector('html').attributes[
+        'data-image-optimizing-origin'
+    ];
+
+    return newOrigin !== oldOrigin;
+};
+
+/**
+ * Array of verfication functions to run to determine
+ * if HTML has changed and needs updation on the UI.
+ */
+const verificationFunctions = [
+    haveResourcesChanged,
+    hasTitleChanged,
+    hasBackendUrlChanged,
+    hasImageOptimizingOriginChanged
+];
+
+/**
+ * Runs all verification functions with new and old
+ * HTML to determine if HTML has changed.
+ *
+ * @param {HTMLElement} newDOM
+ * @param {HTMLElement} oldDOM
+ */
+const hasHTMLChanged = (newDOM, oldDOM) =>
+    verificationFunctions.reduce((acc, fn) => {
```

Comment:
Here again I think you can use the `every` or `some` array functions that fit better than `reduce`.

Reply:
Nice catch. Done 👍